evm/v0.5/contracts/dev/PrepaidAggregator.sol

```diff
@@ -168,8 +172,13 @@ contract PrepaidAggregator is AggregatorInterface, Owned, WithdrawalInterface {
     onlyOwner()
     onlyEnabledAddress(_oracle)
   {
-    oracleCount -= 1;
     oracles[_oracle].endingRound = reportingRoundId;
+    address head = oracleAddresses[oracleCount().sub(1)];
+    uint8 index = oracles[_oracle].index;
+    delete oracles[_oracle].index;
+    oracles[head].index = index;
+    oracleAddresses[index] = head;
+    oracleAddresses.pop();
```

Comment:
Couple things here:
- `head` indicates the front to me, but it's actually the last member. Maybe `tail`?
- I think `oracles[head].index = index` should come before `delete oracles[_oracle].index`, because I think if the oracle was the last one in the list, you'd end up resetting the index right after it's supposed to be deleted.
- This is turning out to be more complicated than I expected, and I'm questioning if it's worth the complexity. Leaning towards worth it, but wanted to throw it out there.

Reply:
- Good with me
- I think a test would have failed if this were the case since we're specifically testing for different positions in the array
- At least it makes for a cool pattern? :man_shrugging:

Reply:
Re 2: I think the storage isn't cleaned up. Is there a test that would catch that?

Reply:
Which storage do you think isn't being cleaned up?

Reply:
`oracles[head].index` when `_oracle` == `head`.

Reply:
It looks like we would just be setting the index to itself then deleting. If we want to test that specifically, we either need to make the `oracles` mapping public or add another getter around getting the `OracleStatus` for a specific oracle.

Reply:
We're deleting it, and then resetting the value to what we just deleted. If you reorder those two lines, this wouldn't be an issue. I don't think the complexity of adding a getter to test this edge case is worth it, but I do think we should fix this edge case where we don't clean up the storage slot. Maybe you could test through gas usage, but I don't fully trust tests like that.

Reply:
That's already been done, the code block from the conversation tab is outdated.

Reply:
My bad, thank you.